macOS/DuckDuckGo/Tab/TabExtensions/AIChatTabExtension.swift

```diff
@@ -175,14 +175,25 @@ final class AIChatTabExtension {
 extension AIChatTabExtension: NavigationResponder {
 
     func decidePolicy(for navigationAction: NavigationAction, preferences: inout NavigationPreferences) async -> NavigationActionPolicy? {
+        // Only handle sidebar, user-initiated, cross-document navigations; otherwise let them proceed normally
         guard isLoadedInSidebar,
               !navigationAction.navigationType.isSameDocumentNavigation,
-              navigationAction.isUserInitiated,
-              let parentWindowController = Application.appDelegate.windowControllersManager.lastKeyMainWindowController
+              navigationAction.isUserInitiated
         else {
             return .next
         }
 
+        // Allow-list: let certain hosts navigate inside the sidebar (e.g., duck.ai)
+        if let host = navigationAction.url.host,
+           host == "duck.ai" || host.hasSuffix(".duck.ai") {
```

Comment:
Couldn't we replace this by checking `navigationAction.url.isDuckAIURL` (I know that it hass a bit broader logic than only duck.ai domain)?